fdb-relational-core/src/main/java/com/apple/foundationdb/relational/recordlayer/query/QueryPlan.java

```diff
@@ -280,12 +281,37 @@ protected <M extends Message> void validatePlanAgainstEnvironment(@Nonnull final
         }
 
         @Nonnull
-        private RelationalResultSet executeExplain(@Nonnull ContinuationImpl parsedContinuation) {
+        protected Optional<RecordQueryPlan> getPlanFromContinuation(@Nonnull ContinuationImpl parsedContinuation,
+                                                                    @Nonnull ExecutionContext executionContext) {
+            final var compiledStatement = parsedContinuation.getCompiledStatement();
+            if (compiledStatement == null || !compiledStatement.hasPlan() || !compiledStatement.hasPlanSerializationMode()) {
+                return Optional.empty();
+            }
+
+            try {
+                final var planSerializationMode = PlanValidator.validateSerializedPlanSerializationMode(
+                        compiledStatement,
+                        OptionsUtils.getValidPlanHashModes(executionContext.getOptions())
+                );
+                final var serializationContext = new PlanSerializationContext(
+                        DefaultPlanSerializationRegistry.INSTANCE, planSerializationMode
+                );
+                return Optional.of(
+                        RecordQueryPlan.fromRecordQueryPlanProto(serializationContext, compiledStatement.getPlan()));
+            } catch (RelationalException ex) {
+                return Optional.empty();
+            }
+        }
+
+        @Nonnull
+        private RelationalResultSet executeExplain(@Nonnull ContinuationImpl parsedContinuation, ExecutionContext executionContext) {
             final var continuationStructType = DataType.StructType.from(
                     "PLAN_CONTINUATION", List.of(
                             DataType.StructType.Field.from("EXECUTION_STATE", DataType.Primitives.BYTES.type(), 0),
                             DataType.StructType.Field.from("VERSION", DataType.Primitives.INTEGER.type(), 1),
-                            DataType.StructType.Field.from("PLAN_HASH_MODE", DataType.Primitives.STRING.type(), 2)),
+                            DataType.StructType.Field.from("PLAN_HASH_MODE", DataType.Primitives.STRING.type(), 2),
+                            DataType.StructType.Field.from("PLAN_HASH", DataType.Primitives.INTEGER.type(), 3),
```

Comment:
We might not have a successful run when the actual plan hash is different from the continuation plan hash. 

Hence, this is redundant in some sense but valuable if the scenarios ever arise.  Having a test for that scenario will be nice!